### DISCUSSION TITLE
Remote connection management for SSH

### DISCUSSION THREAD

**Discussion by mittalyashu** [warpdotdev/Warp#442] [Product ideas]
Option to manage remote connections.

**Comment by elviskahoro** [comment]
Just interested in general, but what do you mean by this? Is there remote management in another terminal or app that I could look at to get an idea?

**Reply by ToontjeM** [reply]
Look at how Wave does this.

**Reply by gitthangbaby** [reply]
or any other terminal. I don't see a point of terminal without SSH.

**Comment by mittalyashu** [comment]
* https://termius.com
* https://coreshell.app

**Reply by Marvinus** [reply]
As well as profiles in Iterm on macos

**Reply by ashebanow** [reply]
Similarly, [Gnome Ptyxis](https://devsuite.app/ptyxis/) adds support for container shells as well. Its nice to be able to automatically get the list of local containers and remote ssh hosts from the system (via docker socket, ssh config, etc).

From a GUI perspective, I think Windows Terminal has the best UX for choosing profiles as part of opening a new tab/window

**Comment by djsipe** [comment]
For my part, it would be nice if there was a way to configure connections to remote servers that you frequently access.  Some means of prefilling the hostname and other connection details that would be part of the `ssh` command.  Then a simple way to be able to pick one of those (maybe from the command pallet?).

**Comment by elviskahoro** [comment]
From Shaqaruden via Discord: https://discord.com/channels/851854972600451112/852249738002956311/943265788779954217
> Not sure if this has been asked for already but I would like to see adding SSH connections to warp to connect to within a couple clicks would be nice. Similar to Termius or Tabby Terminal

**Comment by elviskahoro** [comment]
> Because I usually use SSH to connect to the server remotely for development work, I hope wrap can save server objects (including IP address, user identity, etc.) so that I can easily connect to the server.

**Comment by elviskahoro** [comment]
I've been really liking the work coming out of Charm: https://github.com/charmbracelet
I think we should work on some deep integrations with some of their tools like wishlist and wish.

- https://github.com/charmbracelet/wishlist
- https://github.com/charmbracelet/wish

**Comment by jineshqa** [comment]
Something nice would be similar to profiles in iTerm, which can be configured to SSH into a remote client.

**Comment by michbsd** [comment]
+1
Possibility to add configured aliases into bootstrapped shell.
E.g. I have an alias as such:
alias la='ls -al' 

I would like for Warp to automatically add this alias on all remote hosts I connect to

**Comment by spotashn** [comment]
I think being able to type from the terminal ssh <remotehost> it checks to see if a profile exists for that host and applies it. Much more simple then having to select from a list of saved profiles.

**Comment by elviskahoro** [comment]
https://github.com/brockgr/csshx

**Comment by elviskahoro** [comment]
From @[rana01645](https://github.com/rana01645)
### Describe the solution you'd like?

```shell
It would be great to use local aliases in server.
For instance I use these following aliases in my local for git usage
alias gst='git stash'
alias gsp='git stash pop'
alias gs='git status'
alias gp='git push'

Would be great if I can use this same commands when I ssh to server!
```


### Is your feature request related to a problem? Please describe.

_No response_

### Additional context

_No response_

**Comment by elviskahoro** [comment]
From @neongreen via #1078
### Discord username (optional)

_No response_

### Describe the solution you'd like?

Currently if I'm in an SSH session and I close my laptop, after re-opening it I might get "broken pipe" or just a stuck session.

I'd love if Warp's custom SSH support had auto-reconnect. I know that Mosh exists, but having this built-in without having to install Mosh on the server would be much nicer.


### Is your feature request related to a problem? Please describe.

_No response_

### Additional context

_No response_

**Reply by flaxton** [reply]
I use mosh daily, with tmux over ssh. Formerly on iTerm, now on Warp.

I don't understand why you'd want to duplicate what mosh does, instead of just using it? It's quite easy to use and install, in my experience.

Many times, when I close my laptop on one network, then open it on another, it has killed all of my mosh sessions. 

On iTerm, this does not happen, I can go weeks on many different networks without a disconnection.

Same thing with Blink terminal on my iPad Pro, the connections remain for long periods of time.

So there's something different about the way Warp handles ssh or mosh connections, not sure, but it is annoying.

**Comment by ToontjeM** [comment]
I opened https://github.com/warpdotdev/Warp/issues/1736 and wanted to drop this idea here as well. This is a request for considering implementing SSH multiplexing which makes working in different panes, tabs or windows on remote hosts easier. 
Terminal emulators like iTerm2 or Terminal honor ssh multiplexing configurations. I think 1. Warp needs to honor multiplexing and 2. Warp can enhace it by automatically logging in to the remote host using the connection the new tab was launched from.

**Comment by elviskahoro** [comment]
From #1815
### Discord username (optional)

_No response_

### Describe the solution you'd like?

```text
It would be really nice to have a workflow that supports interactive SSH as a base.

For example, I could define the following workflow commands:


command: |-
    ssh user@machine
    curl http://file.mydomain.tld > /tmp/file
    cat /tmp/file
```

The first step of the workflow would be to create a SSH session and the second and third step would be executed in the SSH session.

Unless I'm mistaken, as the first command is opening an interactive SSH session, the other two commands won't be executed until the first one returns/exits.

Or even better, allow us to use workflows on a remote SSH environment !

Amazing tool by the way!
```


### Is your feature request related to a problem? Please describe.

```text
Just making it easier to reproduce commands on a remote environment.
```


### Additional context

_No response_

### How important is this feature to you?

4

### Warp Internal (ignore) - linear-label:770f6576-d6c0-4e4f-a259-fc64b5156087

_No response_

**Comment by elviskahoro** [comment]
> It would be great if Warp supported drag-and-drop for sending files to the remote host if there is an open SSH connection. In other words, suppose I am connected to my cloud VM via SSH in Warp, and I am looking at a certain directory. If I were to drag a file from my Mac into Warp, it would get transferred to that directory on the cloud VM.

**Reply by dyuzhou** [reply]
example tabby Terminal and WindTerm，Others have a great view on performance.

**Comment by silvewheat** [comment]
For now, I use alias to manage ssh connections.
I write 
`alias launch_server1="sshpass -p 'password' ssh user@IP"`
in ~/.zshrc.
Thus I can type `launch_server1` to connect a remote server.
But it sometimes not work well.
I perfer ssh connection manager like in Tabby.
 
The most annoying thing is, after I use that way to connect ssh. The block not work correctly. There is only one block, I cannot select each single block.

**Comment by zxitedu** [comment]
hi all  Does this need to be implemented in warp now? Or is there a good way to do it? Is there any further follow-up on this issue?

**Reply by elviskahoro** [reply]
this is a whole project that we dont have the bandwidth for atm!

**Comment by alshdavid** [comment]
Just chiming in here not to suggest any specific solution, but in general to mention that my workflow is conducted nearly entirely over ssh.

I have a Linux VM where my work lives. I ssh into it and use VSCode with the SSH extension to conduct remote development.

It's kind of a DIY WSL for MacOS. Having any tooling that can improve the ergonomics of this workflow would be valuable to me

**Comment by jkulak** [comment]
As a side note - I was just looking for a solution I was using 7-9 years ago, also called `warp` and it was a SSH connections manager for CLI. Can not find it now though.

**Comment by dannyneira** [comment]
FR #2144 also requesting this

> would be nice to have something like https://www.termius.com/ where you can more easily manage connections and sessions.

**Comment by AndreaPi** [comment]
Hi! 

As mentioned here https://github.com/warpdotdev/Warp/issues/2106, I think that, for what it concerns remote connections, the most important missing feature in Warp is the capability to detach processes, like `tmux` and `screen`. Being able to detach processes is fundamental whenever you need to run a long process on a remote server (e.g., training a Deep Learning model), thus I often end up having to start `tmux` on the remote server. However, `tmux` multiplexing, copy & paste and mouse mode are much worse than those of Warp. Thus I'd really appreciate if Warp could support detaching processes natively.

**Comment by totoCZ** [comment]
Royal TSX is a good UI example IMO
<img width="552" alt="image" src="https://user-images.githubusercontent.com/5316250/214678684-ee47d85e-e934-4713-87ee-e0b881b41bc4.png">

**Comment by dmz2922990** [comment]
My solution:
![image](https://github.com/warpdotdev/Warp/assets/7579133/5ca0bd97-131b-4fb7-9157-ba941a21aaca)

**Reply by haohanyuzmx** [reply]
but how to ssh with password, i don't want to use sshpass.

**Comment by Haasie** [comment]
+1

**Comment by NedkoHristov** [comment]
+1

**Comment by xudean** [comment]
+1

**Comment by jsonpip** [comment]
+1

**Comment by qiulang** [comment]
@dmz2922990 can you elaborate on how you do that? I am new to warp and I was hoping I could have a list of ssh servers saved and when I click them I just open an ssh session with it. When using iTerm I use profile to do that.

**Reply by snickers2k** [reply]
same here.. it's pretty basic function in default terminal on mac, or not? 
 the only reason i need a terminal , is for remote ssh :D

**Reply by gitthangbaby** [reply]
yep, there's nothing to do locally.. terminal app = ssh. but smart terminals ignore the ssh. Tabby does it correctly, loads ssh config, offers it in the CtrlShiftP popup.  No need to configure anything. After you lock and load all your servers into colored tabs, they will stay there forever, with persistent history, and automatic login each session. Bingo.

**Comment by craigharman** [comment]
+1 from me too. I like the way Tabby handles this.
The only feature missing from Warp for me.

**Comment by sihellem** [comment]
+1

**Comment by bm150** [comment]
+1

**Comment by liuyuan199303** [comment]
+1

**Comment by Alexgolshtein** [comment]
+1

**Comment by NikoS34** [comment]
Putting trendy functions like AI in a terminal is very cool, but basic functions are missing... and a SSH connections manager is one of the main if not the most important ?  What a pity !

**Comment by Skyfay** [comment]
+1

**Comment by defcon8** [comment]
+2

**Comment by jravolio** [comment]
+1

**Comment by uncre4tive** [comment]
+1

**Comment by songispm** [comment]
+1

**Comment by rakosolapov** [comment]
+1

also you can save command as a workflow:
`ssh -t myusername@18.218.45.121 "cd /opt/myfolder && sudo su"`
-t stands for force pseudo-tty allocation (to execute code in quotes on remote server)

**Comment by rithviknishad** [comment]
+1

**Comment by lopongoaca** [comment]
+1

**Comment by jyncus** [comment]
+1

**Comment by highdeserthacker** [comment]
+1

**Comment by eka-m** [comment]
+1

**Comment by joaberg** [comment]
I think it should be pretty easy to implement.  
Just use the .ssh/config and the zsh-ssh plugin from https://github.com/sunlei/zsh-ssh   
If you can get the plugin working, and make it launchable with keybind, that would be awesome.  

It should look like this:
![image](https://github.com/warpdotdev/Warp/assets/58996735/3e353873-bb3b-4fb8-b3b6-fa30832e18fc)

The connections from the config does show up in the suggestions, but it cant look or search in the config descriptions etc.  
![image](https://github.com/warpdotdev/Warp/assets/58996735/c80600b2-8596-49fe-bdb4-56b1208eb27a)

**Comment by d86e** [comment]
+1

**Comment by tmg0** [comment]
+1

**Comment by zsinryu** [comment]
+1

**Comment by sfkmk** [comment]
+1

**Comment by PiotrRaszkowski** [comment]
+1

**Comment by nk3750** [comment]
+1

**Comment by rikgirbes** [comment]
+1

**Comment by lukaskretzschmer** [comment]
With so much data collection and VC funding pressure I think I'll just use Termius for now :/

**Comment by wkepting** [comment]
+1

**Comment by zsxkib** [comment]
+=1

**Comment by linktomarkdown** [comment]
+1

**Comment by adit1o** [comment]
+1000 i love warp, and i play with SSH for working everyday. please add this feature

**Comment by aofodo** [comment]
+1

**Comment by wenslo** [comment]
+1

**Comment by PickX33** [comment]
+1

**Comment by indoyoon** [comment]
I ended up switching to Termius

**Comment by k0k** [comment]
+1

**Comment by pwinntii** [comment]
I was really surprised to realize this isn't supported! It is such a core part of iTerm that it took me a couple of days to realize it was missing from Warp. 

I have three (3) Profiles saved in iTerm: 
1. "Default" has Command set to "Login shell" 
2. "WD NAS" has Command set to "ssh" with the hostname as the argument, and ssh integration turned off
3. "OctoNAS" has Command set to "ssh" with the argument "-p 323 octonas-proxy" and ssh integration turned off. 

The last used to be "command" instead of "ssh," but I realized while typing this up that I could switch it, so I did. 

I was really looking forward to using Warp on my home Mac as well as my work Mac, but without support for ssh, I guess I'll stick with Warp at work and iTerm at home.

**Comment by dzqdzq** [comment]
+1

**Comment by NaveedKhan66** [comment]
+1

**Comment by mrsaeedi93** [comment]
+1
would love to see this feature in warp

**Comment by woaiso** [comment]
+1

**Comment by scalesops** [comment]
+1

**Comment by rikgirbes** [comment]
It’s been over three years since this feature request was opened, and it’s still not implemented. A simple SSH host list would be incredibly useful, even without authentication support (most users rely on key-based authentication anyway). This seems like a straightforward addition that could really improve the user experience. Can't we prioritize this?..

**Comment by keroro520** [comment]
+1

**Comment by gghidoni** [comment]
+1

**Comment by vincent-vinf** [comment]
+1

**Comment by vincent-vinf** [comment]
We really need the ssh connection management feature, I'm using tabby right now to manage connections to dozens of machines, and this feature has made me much more productive by eliminating the need to memorize the ip, connection credentials, and purpose of a machine. Adding this feature to warp would greatly improve the competitiveness of the product.
In addition, tabby's jump host function is also very useful!
<img width="792" alt="image" src="https://github.com/user-attachments/assets/8e10226e-6b3c-4431-b676-c22e713b4b13">

**Reply by dylandacosta8** [reply]
This is exactly what Warp needs. Hope someone from the dev team takes a look at Tabbys SSH management

**Comment by GaborTorma** [comment]
+1

**Comment by qiulang** [comment]
It is quite annoying that I now use 3 terminal tools, **warp** mainly because I like it let me mouse positioning anywhere in command input, **Termius** for ssh management and easy sftp, **iTerm2** as I have many profiles set already and I still need them.

So far **Termius**  is my favorite but it does not show unicode character correctly, which seems to be an easy bug to fix but them did not.

**Comment by dereknorrbom** [comment]
+1

**Comment by keroro520** [comment]
SSH management is important, please.

**Comment by The-Man11** [comment]
🙌🏼

**Comment by AfshinHonari** [comment]
+1

**Comment by nuno4793** [comment]
+1

For me I would love to be able to have something similar to the Windows Terminal Profiles which allow me to leverage my .ssh/config for.

Without this me or my other colleagues couldn't ever adopt Warp terminal

**Comment by damago1** [comment]
I have installed terminus and immediately I started to look up how to save a list of my ssh connections. I thought that just the user interface is so weird that I do not understand something. For me such option is SO OBVIOUS that I could not believe that it does not exist. It is a terminal. You use terminal to connect to machines. Most warp users are users who do a bit more than type single ls/dir, and usually work with different machines.

**Comment by pwinntii** [comment]
It has been more than three years. This is the fifth-highest-voted of All Time, the highest-voted by far under Product Ideas, and has had no response I've seen. It feels like the Warp team cares more about monetizable AI features than basic functionality used by the majority of terminal users. 

I'm out. I've tried to make this work for years now, and it's just as clumsy today as when I started. Maybe slightly more clumsy, given that the AI features have gotten more intrusive. 

I'm uninstalling Warp from my three macs and going back to iTerm 2 full-time. Good luck, Warp team.

**Reply by earmellin** [reply]
I'm very tempted to do the same

**Reply by lopongoaca** [reply]
I'll to do the same, uninstalls wrap and install Ghostty.

**Reply by rithviknishad** [reply]
I've already switched to ghostty couple weeks back

**Reply by nuno4793** [reply]
I am very tempted to do the same tbh, I have tried but all these very basic functionality being missing is kinda a no no.

In my case I tried Windows and Linux.

**Comment by ashebanow** [comment]
I’m curious what missing functionality you consider essential in kitty,
ghostly, etc. I really like warp’s blocks, command line editing, etc, but I
don’t consider those essential.

**Reply by nuno4793** [reply]
For me, it is the simple/basic things that Warp doesn't have. A the moment the deal break is the inability to have "profiles" for ssh connections,  something like Windows Terminal [wt profiles](https://pureinfotech.com/create-duplicate-profiles-windows-terminal/)

This already made me stop using it when I tried back a few months in WSL, I decided to give it another go with the Windows version and I didn't last two days on it. I found myself going back to Windows Terminal.

Don't get me wrong, Warp is amazing, what the guys did is no small achievement and, but I don't really understand how something like this is not present.

**Comment by ashebanow** [comment]
Oh, ok, gotcha. I thought you meant that kitty/iterm/etc were missing
critical features. My bad.

**Comment by ashebanow** [comment]
By the way, for those watching the bug, you can sort of emulate this functionality using launch configurations. It isn't nearly as convenient as the popup in the window titlebar, but it works.

**Reply by nuno4793** [reply]
@ashebanow ,

Being honest I did try to do that but couldn't figure it out. Also, I created a Launch configuration, and I can't see it from warp. Maybe it's me doing something wrong but it would be interesting to have a more detailed example in the docs.

**Comment by hilsonp** [comment]
The only reason I open a local terminal is to run git or brew commands. Sometimes, rarely, I'll open it to do fancy text treatment or work with curl an jq.
The rest of the day, I spend hours in a terminal window managing complete networks.

To do so efficiently, I need a connection manager that let me:
- put connections in a tree structure
- give meaningful names to the connections (and not simply the hostname as those can be cryptic)
- give an icon or at least a color (this helps me differentiate environments (DEV vs TEST vs PROD for example))
- be able to force the pane/tab to respect the connection name/color

I always had this in all terminal emulators I used.
I've been searching for a while where it was "hidden" in Warp to finally come to the conclusion that it does not exist at all.

Warp is mind blowing for anything else but this was such a surprise for me.
This is a mandatory basic feature I need to manage all my customers hosts and I'll have to wait that warp offers something on this side before I can really use it.

Thank you for the great work !

**Reply by earmellin** [reply]
done like this it would be absolutely perfect :)

**Reply by HuweyII** [reply]
This is a great list. It would be nice if that color (to differentiate DEV/TEST/PROD) could continue to a settable color scheme for the connection.   In SecureCRT, I will set Production servers with a different color so that I don't accidentally send them commands when I have sessions open to them for some reason.

**Comment by deku0818** [comment]
Hope to be online soon!!!!

**Comment by HuweyII** [comment]
I've only used Warp for a few weeks, and it's pretty great.  But as long as it doesn't have this sort of feature, it will never be my primary terminal.  Perhaps an SSH connection manager will get added before the next time I need to buy SecureCRT, and I can give my money to warp instead.  :)

**Comment by sbfkcel** [comment]
Urgently needed, otherwise it will be an incomplete terminal.

**Comment by a7medo778** [comment]
definatly needed, but i suggest a more friendly GUI to add the SSH creds

**Comment by betasecond** [comment]
Nice Warp,update on any progress or plans regarding this functionality?

**Comment by earmellin** [comment]
a lot of AI planned for the new update but no plan for this basic feature?

**Comment by rkondra-eightfold** [comment]
@earmellin 
https://docs.warp.dev/terminal/warpify/ssh
also: check my setup for ET: https://github.com/warpdotdev/Warp/discussions/501#discussioncomment-13434422

**Reply by earmellin** [reply]
I'll take a look, thanks! but my idea was closer to a connection manager like Termius or at least like in iTerm

**Comment by xvoron** [comment]
+1

**Comment by gabrielbryk** [comment]
+1

**Comment by Ixe1** [comment]
This would be really useful, +1 here too, I mainly use Warp for SSH stuff but it's a pain that it doesn't have the capability built-in.

**Comment by joelgaribaygz** [comment]
I manage several VPS via SSH and it would be helpful to have this active option! My +1 goes

**Comment by IsaacHatilima** [comment]
One more user needing this.

**Comment by mightywomble** [comment]
+1 for this,

**Reply by mightywomble** [reply]
I wrote this as a stopgap: https://github.com/mightywomble/sshtui
Not perfect

**Comment by DDZ-DO** [comment]
+1

**Comment by kodewulf** [comment]
Would be nice if basic terminal features are working before trying to cram AI into a terminal app and monetizing the hell out of it.  Basic SSH connection profile management has been requested about 4 years ago and it's still not a feature.  Here's a suggestion: use your much publicized AI features to create the code to add profile management to the application.  At least we'll know that your AI is useful then.

**Reply by rikgirbes** [reply]
I completely agree. I’ve been waiting for this for over a year, and in the meantime, I switched back to iTerm2, which has this built in and even includes AI features now. Honestly, the only thing I really use AI for is basic stuff like “write an ffmpeg command to convert this to that,” so simple prompt capabilities are more than enough for me.
Warp used to be quite nice, but now it just feels like it’s trying to force AI down my throat.

**Comment by NikoS34** [comment]
On my side the request for this basic function dates from the beginning of 2024... and since then absolutely no progress or roadmap on the subject ? and like you back to Iterm2 and very satisfied !

**Reply by earmellin** [reply]
I also plan to go back to iTerm2, they bloated Warp with AI features that I simply don't use, because I prefer to do AI assisted development in my IDE, leaving behind basic functionality that any ssh terminal should have

**Comment by a7medo778** [comment]
Seriously its almost unusable having to tell it to connect to a vps every single time

ahmad

**Comment by ashebanow** [comment]
Almost unusable? I want this feature too, as a convenience, but isn't that
a little dramatic?

**Comment by a7medo778** [comment]
Not really, you ask to run it on a remote server

Then the ssh command
Then the pass
Then if you leave the terminal a bit it disconnect and revert back to the local machine
And the cycle goes again

ahmad

**Comment by ashebanow** [comment]
Off topic, but if you are doing all that, you are doing it the hard way.
Use per-host or per-domain ssh certificates, not passwords -its way, way
more secure. Put an entry in your .ssh/config to tell it which user
account, which cert, and which machine to connect to when you say 'foo'.
Then the command to connect is literally 'ssh foo' and you are done.

**Comment by a7medo778** [comment]
Thanks for the tip, will try this today 👍

ahmad

**Comment by ieuze** [comment]
This issue has gone on its 4th anniversary and the feature is still in the air

**Reply by gabrielbryk** [reply]
laughed out loud

**Comment by K3UL** [comment]
Warp is nice, but a terminal with no SSH sessions/profiles is just not acceptable.... I think this one of the only ones on the market, definitely the only one beyond the "main" players.  
Personally I am switching back to iterm2.  

See you again if some day you decide to prioritize the terminal stuff over the AI stuff

**Reply by cedric-marcoux** [reply]
> Warp is nice, but a terminal with no SSH sessions/profiles is just not acceptable.... I think this one of the only ones on the market, definitely the only one beyond the "main" players. Personally I am switching back to iterm2.
> 
> See you again if some day you decide to prioritize the terminal stuff over the AI stuff

same

**Comment by crispybegs** [comment]
googled this thinking i must be going mad there's no connection manager in warp but no, here we are still

**Comment by SheepDomination** [comment]
A SSH connection manager would be great, especially when connecting to multiple servers using SSH. I'd also like to see some form of SSL encryption management which ties the SSL to Warp but doesn't store the encryption only what is important from the key.

**Comment by lloydbayley** [comment]
I must say, that since the OP's post in 2021, I'm losing hope. I'm still not sure what constitutes a good business-decision with this current model. If a CM was added, there'd be a lot of happy people and potential custom. Still, 5 years shows us the way things are. I'd jump at it if I was the developer but bad decisions have to come from somewhere, I suppose! Gives us something to whinge about anyway! :)

**Comment by laneyweb** [comment]
I might not understand how or where to look but is this Feature Request in state of accepted and we are waiting or is it just left here in a discussion? Is there a way to see what the Devs are adding to thier backlog?

**Comment by danmartuszewski** [comment]
I had been waiting for it for so long and finally created https://github.com/danmartuszewski/hop that works beautifully with Warp

**Comment by xjiun** [comment]
+1, this is the single feature that keeps me on Termius instead of Warp.

I manage 30+ servers across multiple environments (prod / staging / dev) and customers, and the workflow I need is simply:

- save host / port / user / identity file / jump host as a profile
- group them by env or project, with tags
- one-click launch from a sidebar — no retyping `ssh -i ... -J ... -p ...`
- ideally sync across machines

`~/.ssh/config` only covers part of it — there's no GUI list, no grouping, no per-host metadata, and no fast launcher. iTerm2's profiles + Termius's Hosts both solve this in different ways; Warp currently solves neither.

I understand the team's focus is "Warpify the remote shell," but for a lot of us the pre-step (organizing and launching the connection) is the actual daily friction. Warpify only matters after I've connected — and right now I'm connecting in Termius.

Happy to switch the day this lands. Posting here per maintainer guidance instead of opening another duplicate issue.

**Reply by danmartuszewski** [reply]
@xjiun You can do this with https://github.com/danmartuszewski/hop
If you found sth missing, open a ticket please

**Reply by qiulang** [reply]
I also use Termius and iTerm2, mainly Termius because it can connect to remote ssh servers with one click, which is much better than iTerm2's profile solution. But the start up command feature in Termius is a paid feature and I think per 10$ per month for that is ridiculous. 

When I connect to the remote ssh server, 50% of time I will start a tmux  session, iTerm2's CC feature is the main reason I still use iTerm2.

I have to say missing these two killer features outweighs all the other nice‑to‑have features warp provides.

**Comment by ashebanow** [comment]
Now that they've gone open source, someone has the opportunity to bey a
hero...

**Reply by SheepDomination** [reply]
Now that it's Open Source it also open the door for others to steal the code and add it to their terminal. 
I suppose all that we can hope for is some of the issues back logged to be fixed or added sooner then later.

**Comment by ankursarv** [comment]
If you guys like termius like UI - for macos we developed one terminal.

https://github.com/Sarv/SarvTerminal
